js/script.js

```diff
@@ -0,0 +1,117 @@
+$(function() {
+    function randomString() {
+		var chars = '0123456789abcdefghiklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXTZ';
+		var str = '';
+		for (var i = 0; i < 10; i++) {
+			str += chars[Math.floor(Math.random() * chars.length)];
+		}
+		return str;
+	}
+
+	function Column(name) {
+		var self = this;
+		this.id = randomString();
+		this.name = name;
+		this.$element = createColumn();
+
+			function createColumn() {
+				// CREATING COMPONENTS OF COLUMNS
+				var $column = $('<div>').addClass('column');
+				var $columnTitle = $('<h2>').addClass('column-title').text(self.name);
+				var $columnCardList = $('<ul>').addClass('column-card-list');
+				var $columnDelete = $('<button>').addClass('btn-delete').text('x');
+				var $columnAddCard = $('<button>').addClass('add-card').text('Add a card');
+
+				// ADDING EVENTS
+				$columnDelete.click(function() {
+					self.removeColumn();
+				});
+				$columnAddCard.click(function(event) {
+					self.addCard(new Card(prompt("Enter the name of the card")));
+				});
+
+				// CONSTRUCTION COLUMN ELEMENT
+				$column.append($columnTitle)
+					.append($columnDelete)
+					.append($columnAddCard)
+					.append($columnCardList);
+
+				// RETURN OF CREATED COLUMN
+				return $column;
+			}
+		}
+		
+		Column.prototype = {
+			addCard: function(card) {
+			  this.$element.children('ul').append(card.$element);
+			},
+			removeColumn: function() {
+			  this.$element.remove();
+			}
+		};
+		
+	function Card(description) {
+		var self = this;
+
+		this.id = randomString();
+		this.description = description;
+		this.$element = createCard();
+
+			function createCard() {
+				// CREATING THE BLOCKS
+				var $card = $('<li>').addClass('card');
+				var $cardDescription = $('<p>').addClass('card-description').text(self.description);
+				var $cardDelete = $('<button>').addClass('btn-delete').text('x');
+
+				// BINDING TO CLICK EVENT
+				$cardDelete.click(function(){
+					self.removeCard();
+				});
+
+				// COMBINING BLOCKS AND RETURNING THE CARD
+				$card.append($cardDelete)
+					.append($cardDescription);
+				return $card;
+			}
+			
+		Card.prototype = {
+			removeCard: function() {
+				this.$element.remove();
+				}
+			}
+		}
+
+	var board = {
+		name: 'Kanban Board',
+		addColumn: function(column) {
+			this.$element.append(column.$element);
+			initSortable();
+		},
+		$element: $('#board .column-container')
+	};
+	
+	function initSortable() {
+		$('.column-card-list').sortable({
+			connectWith: '.column-card-list',
```

Comment:
try to move card to empty column :) fix it

Reply:
function initSortable() looks good, please show me what I have to do.

Reply:
in css you should set a min-height for column, you can't move card to column that doesn't exist